src/transformStateWithClones.js

```diff
@@ -6,8 +6,40 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
-}
+  let objectKeeper = { ...state };
+  const stateModified = [];
+
+  for (const index of actions) {
+    if (index.type === 'addProperties') {
+      const localTemporaryObject = { ...objectKeeper };
```

Comment:
You create unnecessary copy here. You push copied object to `stateModified` anyway.